test/LRPTests/APITests.cpp

```diff
@@ -0,0 +1,164 @@
+
+            if (wasProcessCreated == FALSE)
+            {
+                VERIFY_SUCCEEDED(GetLastError());
+            }
+
+            // Wait for the process to come up and be captured in the snapshot from verification step.
+            Sleep(1000);
+            VerifyLRP_IsRunning(true);
+        }
+
+        void VerifyLRP_IsRunning(bool isRunning)
+        {
+            wil::unique_handle processesSnapshot(CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0));
+            VERIFY_IS_NOT_NULL(processesSnapshot.get());
+
+            PROCESSENTRY32 processEntry{};
+            processEntry.dwSize = sizeof(processEntry);
+
+            BOOL result{ Process32First(processesSnapshot.get(), &processEntry) };
+            while (result != FALSE)
+            {
+                if (wcscmp(L"PushNotificationsLongRunningTask.exe", processEntry.szExeFile) == 0)
+                {
+                    VERIFY_IS_TRUE(isRunning);
+                    DWORD processId{ processEntry.th32ProcessID };
+
+                    wil::unique_handle longRunningProcessHandle(OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, processId));
+                    DWORD exitCode{};
+                    BOOL exitCodeProcess{ GetExitCodeProcess(longRunningProcessHandle.get(), &exitCode) };
+
+                    VERIFY_SUCCEEDED(exitCodeProcess == FALSE ? GetLastError() : S_OK);
+                    VERIFY_ARE_EQUAL(exitCode, STILL_ACTIVE);
+
+                    return;
+                }
+
+                result = Process32Next(processesSnapshot.get(), &processEntry);
+            }
+
+            VERIFY_IS_FALSE(isRunning);
+        }
+
+        TEST_METHOD(LaunchLRP_FromCoCreateInstance)
+        {
+            winrt::com_ptr notificationPlatform{ winrt::try_create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL) };
+            VERIFY_IS_NOT_NULL(notificationPlatform.get());
+        }
+
+        TEST_METHOD(RegisterUnregisterLongRunningActivator)
+        {
+            winrt::com_ptr notificationPlatform{ winrt::try_create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL) };
+            VERIFY_IS_NOT_NULL(notificationPlatform.get());
+
+            wil::unique_cotaskmem_string lrpAppId;
+            VERIFY_NO_THROW(notificationPlatform->RegisterFullTrustApplication(c_processName, winrt::guid(), &lrpAppId));
+            VERIFY_NO_THROW(notificationPlatform->RegisterLongRunningActivator(c_processName));
+            VERIFY_NO_THROW(notificationPlatform->UnregisterLongRunningActivator(c_processName));
+            VERIFY_NO_THROW(notificationPlatform->UnregisterFullTrustApplication(c_processName));
+        }
+
+        TEST_METHOD(RegisterUnregisterLongRunningActivatorWithClsid)
+        {
+            winrt::com_ptr notificationPlatform{ winrt::try_create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL) };
```

Comment:
this is repeat thing can we move it to a common setup?

Reply:
Yeah, I'll move this to a common function.